docs/end_to_end_tutorials/principled_dev_practices.md

```diff
@@ -0,0 +1,42 @@
+# Principled Development Practices
+
+In order to develop your application, it can help to implement some principled development practices.
+
+We've boiled down the key practices into 3 pillars.
+
+We will also try to do a compare and contrast with traditional software engineering and ML development, highlighting similarities and differences.
+
+Here are the main pillars of principled development of LLM and RAG applications:
+
+```{toctree}
+---
+maxdepth: 1
+---
+/end_to_end_tutorials/dev_practices/observability.md
+/end_to_end_tutorials/dev_practices/evaluation.md
+/end_to_end_tutorials/dev_practices/monitoring.md
+```
+
+## The Development Pathway
```

Comment:
high-level comment related to @logan-markewich's point, I'm not actually sure what the goal of these below sections are. Why are these below the 3 sub pages linked above and what's the high-level thing you're trying to convey - is it common challenges and how to fix them?

Reply:
It's information about the general concerns the user may have when developing their application.

Reply:
at a high-level i'm still trying to figure out how to relate these sections below to the sections of observability/evaluation/monitoring that you've filled out. Just as a user what is my journey through these docs. is it i go through this page, read the challenges, then go into observability/evaluation/monitoring and figure out the solution? or i read those pages first and then do a back and forth between this page and more specific pages.

thoughts? we don't need to have a complete solution before we publish but wanted to arrive at an understanding

Reply:
I thought it might be awkward at first but honestly having the links at the top and more elaboration down below worked better for me, cuz I'd click on the links after skimming what's down below, but if it's a giant wall of text I might not even get to the links.

Reply:
I do agree that having the links below again would be nice. 

Or perhaps having a separate page for the more in-depth explanations could be better, just didn't want it to not be discoverable as it seems generally helpful

Reply:
yeah i agree links should come before text. just wondering if there's ways to link this text / challenges into the three pillars so there's a bit of a relationship there